### PR TITLE
feat(textinput): apply the new example layout to the textinput examples

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

Applying the new example layout to the TextInput component examples.

To test, simply navigate to the `Text Input` section in the plasma website.

### Potential Breaking Changes

None, only affects the website

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by aiheon** [comment]
Since it takes some time for the sandox to load I think we could try and use the plasma loading to hide the empty div? :eyes: 
What do you guys think?

File: packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx [hunk 1]
```diff
@@ -1,105 +1,106 @@
...
+                <Button
+                    name='Change dish for "sushis"'
+                    enabled={state.value !== 'sushis'}
+                    onClick={() => dispatch({type: 'change-value', payload: 'sushis'})}
+                />
+                <Button
+                    name="Show validation"
+                    onClick={() => {
+                        dispatch({type: 'show-validation'});
+                    }}
```
Comment (toofff) [inline packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx:64]:
```suggestion
                    onClick={() => dispatch({type: 'show-validation'})}
```

File: packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx [hunk 2]
```diff
@@ -1,105 +1,106 @@
...
+                />
+                <Button
+                    name="Hide validation"
+                    onClick={() => {
+                        dispatch({type: 'hide-validation'});
+                    }}
```
Comment (toofff) [inline packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx:70]:
```suggestion
                    onClick={() => dispatch({type: 'hide-validation'})}
```

File: packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx [hunk 3]
```diff
@@ -1,105 +1,106 @@
...
-        </>
+            <CurrentState />
+            <InteractiveButtons />
+        </FormProvider>
     );
-};
+`;
+
+// start-print
+export const TextInputExamples: React.FunctionComponent = () => (
```
Comment (aiheon) [inline packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx:97]: nice and tidy :smile:

**Review by aiheon — APPROVED** [review]

File: packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx [hunk 4]
```diff
@@ -1,105 +1,106 @@
...
-                        />
-                    </div>
-                    <HookUsageDemo />
-                </Section>
-            </Section>
+            <TextInput
+                required
+                showValidationOnBlur
+                validate={nonEmptyValidation}
+                type="text"
```
Comment (toofff) [inline packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx:27]: The type `text` could not be a default value?

File: packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx [hunk 5]
```diff
@@ -1,105 +1,106 @@
...
+        const isEmpty = !value;
+        const isWhiteSpace = !isEmpty && !value.trim();
+        if (isWhiteSpace) {
+            return {status: 'warning', message: 'A white space is not empty, but is not ideal.'};
+        } else if (isEmpty) {
+            return {status: 'invalid', message: 'Cannot be empty.'};
+        } else {
+            return {status: 'valid'};
+        }
+    };
```
Comment (toofff) [inline packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx:19]:
```suggestion
    const nonEmptyValidation: InputValidator = (value: string) => {
        const isEmpty = !value;
        const isWhiteSpace = !isEmpty && !value.trim();
        if (isWhiteSpace) {
            return {status: 'warning', message: 'A white space is not empty, but is not ideal.'};
        }

        if (isEmpty) {
            return {status: 'invalid', message: 'Cannot be empty.'};
        }

        return {status: 'valid'};
    };
```

**Review by toofff — APPROVED** [review]

File: packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx (hunk 4 above)
Comment (gdostie) [reply packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx:27]: When implementing the TextInput, I thought about this and decided not to give it a default value. The reason is that people will not think about it and leave it as `text` for emails or urls, etc. Making it required forces us to think about which type it should have.

**Review by GermainBergeron — APPROVED** [review]

**Pull request merged**